### PR TITLE
Qute - improve content type detection

### LINKED ISSUE #10991: Qute: Unknown media type for suffix: .graphql
**Description**
I am using `.graphql` files to produce valid GraphQL content from Qute templates (see the .txt files in https://github.com/quarkusio/quarkus-backports/tree/master/src/main/resources/templates/GitHubService). If I set their suffix to `.graphql`, Qute still finds them and parses correctly but the following logging message keeps appearing in the logs: 

```
2020-07-27 09:35:38,530 WARN  [io.qua.qut.run.TemplateProducer] (Quarkus Main Thread) Unknown media type for suffix: .graphql
```

**Implementation ideas**
Make the log a debug statement or have some way to specify custom content types

### PR DESCRIPTION
**Pull request by mkouba**
- add config property for additional mappings
- resolves #10991

### CODE REVIEW

**Comment by mkouba** [comment]
@gastaldi you would add `quarkus.qute.content-types.graphql=application/graphql` to get rid of the warning, plus you should also specify `quarkus.qute.suffixes=qute.html,qute.txt,html,txt,graphql` to get full support ;-)

**Comment by gsmet** [comment]
Looks like a nice addition for 1.7. @gastaldi I let you review it?

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/ContentTypes.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if (dotIdx != -1) {
+            String suffix = fileName.substring(dotIdx + 1, fileName.length());
+            String additionalContentType = config.contentTypes.get(suffix);
+            if (additionalContentType != null) {
+                return additionalContentType;
+            }
+            if (suffix.equalsIgnoreCase("json")) {
+                return Variant.APPLICATION_JSON;
+            }
+            String contentType = URLConnection.getFileNameMap().getContentTypeFor(fileName);
```
Comment (gastaldi) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/ContentTypes.java:41]: We could use `Files.probeContentType` too (or instead?) to take advantage of libraries that register their own `java.nio.file.spi.FileTypeDetector` through the ServiceLoader mechanism

**Review by gastaldi — APPROVED** [review]
Excellent! Added one minor optional comment

**Comment by gsmet** [comment]
Merging this one. We can still improve on @gastaldi 's suggestion later.

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/ContentTypes.java (hunk 1 above)
Comment (mkouba) [reply extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/ContentTypes.java:41]: I was thinking about `Files.probeContentType()` too but given the fact that the result may differ, based on the OS or presence of other libraries, I decided to choose the simple and straightforward solution for now. Also it seems that the default `FileTypeDetector` impl is makes use of `URLConnection.getFileNameMap()` anyway...

**Pull request merged**